src/RwaRegistry.sol

```diff
@@ -0,0 +1,563 @@
+        emit AddSupportedComponent("liquidationOracle");
+
+        _supportedComponents.add("outputConduit");
+        emit AddSupportedComponent("outputConduit");
+
+        _supportedComponents.add("inputConduit");
+        emit AddSupportedComponent("inputConduit");
+
+        _supportedComponents.add("jar");
+        emit AddSupportedComponent("jar");
+
+        _supportedComponents.add("jarInputConduit");
+        emit AddSupportedComponent("jarInputConduit");
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    /*//////////////////////////////////
+              Authorization
+  //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /*//////////////////////////////////
+     Supported Components Management
+  //////////////////////////////////*/
+
+    /**
+     * @notice Adds a supported component name to the registry.
+     * @dev Adds a new type of component that should be supported.
+     * @param name The "pascalCased" name of the component.
+     */
+    function addSupportedComponent(bytes32 name) external auth {
+        if (_supportedComponents.contains(name)) {
+            revert ComponentAlreadySupported(name);
+        }
+
+        _supportedComponents.add(name);
+
+        emit AddSupportedComponent(name);
+    }
+
+    /**
+     * @notice Returns whether a component name is supported or not.
+     * @param name The "pascalCased" name of the component.
+     * @return Whether a component name is supported or not.
+     */
+    function isSupportedComponent(bytes32 name) external view returns (bool) {
+        return _supportedComponents.contains(name);
+    }
+
+    /**
+     * @notice Lists the names of all types of components supported by the registry.
+     * @return The list of component names.
+     */
+    function listSupportedComponents() external view returns (bytes32[] memory) {
+        return _supportedComponents.values();
+    }
+
+    /*//////////////////////////////////
+     Deals & Components Management
+  //////////////////////////////////*/
+
+    /**
+     * @notice Adds a deal identified by `ilk` to the registry.
+     * @param ilk The ilk name.
+     */
+    function add(bytes32 ilk) external auth {
+        _addDeal(ilk);
+    }
+
+    /**
+     * @notice Adds a deal identified by `ilk` with its components to the registry.
+     * @param ilk The ilk name.
+     * @param names The list of component names.
+     * @param addrs The list of component addresses.
+     * @param variants The list of component variants.
+     */
+    function add(
```

Comment:
Following the above on `ilk-registry` interface compliance, I don't see here a behaviour to have a permissionless update of a specific component that can be query on-chain by the function to reduce costly operations via spells.

Reply:
There is no reliable on-chain source of information to allow for a permissionless update.
Actually, the RWA Registry aims to be THE source of truth for RWA deals, therefore it must be trusted.